src/types/error/mod.rs

```diff
@@ -139,199 +134,86 @@ impl<'r> Responder<'r> for AppError {
-            AppErrorKind::TooManyRequests => Status::TooManyRequests,
-            AppErrorKind::ComplaintError { .. }
-            | AppErrorKind::BounceSoftError { .. }
-            | AppErrorKind::BounceHardError { .. } => Status::TooManyRequests,
-            AppErrorKind::BadRequest | AppErrorKind::InvalidEmailParams => Status::BadRequest,
-            AppErrorKind::MissingEmailParams(_) => Status::BadRequest,
+            AppErrorKind::InvalidPayload(_) => Status::BadRequest,
+            AppErrorKind::Complaint { .. } => Status::TooManyRequests,
+            AppErrorKind::SoftBounce { .. } => Status::TooManyRequests,
+            AppErrorKind::HardBounce { .. } => Status::TooManyRequests,
             _ => Status::InternalServerError,
         }
     }
 
     pub fn errno(&self) -> Option<u16> {
         match self {
-            AppErrorKind::RocketError(_) => Some(100),
-
-            AppErrorKind::MissingEmailParams(_) => Some(101),
-            AppErrorKind::InvalidEmailParams => Some(102),
-
-            AppErrorKind::InvalidProvider(_) => Some(103),
-            AppErrorKind::ProviderError { .. } => Some(104),
-            AppErrorKind::EmailParsingError(_) => Some(105),
-
-            AppErrorKind::ComplaintError { .. } => Some(106),
-            AppErrorKind::BounceSoftError { .. } => Some(107),
-            AppErrorKind::BounceHardError { .. } => Some(108),
-
-            AppErrorKind::AuthDbError(_) => Some(109),
-
-            AppErrorKind::QueueError(_) => Some(110),
-            AppErrorKind::InvalidNotificationType => Some(111),
-            AppErrorKind::MissingNotificationPayload(_) => Some(112),
-
-            AppErrorKind::MissingSqsMessageFields { .. } => Some(113),
-            AppErrorKind::SqsMessageHashMismatch { .. } => Some(114),
-            AppErrorKind::SqsMessageParsingError { .. } => Some(115),
-
-            AppErrorKind::DurationError(_) => Some(116),
-            AppErrorKind::DbError(_) => Some(117),
-
-            AppErrorKind::NotImplemented => Some(118),
-
-            AppErrorKind::HmacError(_) => Some(119),
-
-            AppErrorKind::JsonError(_) => Some(120),
-
-            AppErrorKind::BadRequest
-            | AppErrorKind::NotFound
-            | AppErrorKind::MethodNotAllowed
-            | AppErrorKind::UnprocessableEntity
-            | AppErrorKind::TooManyRequests
-            | AppErrorKind::InternalServerError => None,
+            AppErrorKind::Internal(_) => Some(100),
+            AppErrorKind::NotImplemented(_) => Some(101),
+            AppErrorKind::InvalidPayload(_) => Some(102),
+            AppErrorKind::InvalidNotification(_) => Some(103),
+            AppErrorKind::QueueError(_) => Some(104),
+            AppErrorKind::InvalidDuration(_) => Some(105),
+            AppErrorKind::Complaint { .. } => Some(106),
+            AppErrorKind::SoftBounce { .. } => Some(107),
+            AppErrorKind::HardBounce { .. } => Some(108),
```

Comment:
These last three are the ones that have been preserved to match their old values, in case you want to review that.